src/cljs/main/broadfcui/main.cljs

```diff
@@ -175,9 +175,32 @@
    (fn []
      {:user-status #{}})
    :component-will-mount
-   (fn [{:keys [this]}]
+   (fn [{:keys [this state]}]
      (init-nav-paths)
-     (this :handle-hash-change))
+     (this :handle-hash-change)
+     (set! js/window.forceSignedIn
+           (fn [auth-token]
+             (utils/ajax {:url (str "https://www.googleapis.com/oauth2/v3/tokeninfo?access_token=" auth-token)
+                          :on-done
+                          (fn [{:keys [status-code success? get-parsed-response]}]
+                            (if success?
+                              (let [{:keys [email sub]} (get-parsed-response)
+                                    instance (clj->js
+                                              {:currentUser
+                                               {:get
+                                                (fn []
+                                                  (clj->js {:getAuthResponse
+                                                            (fn [] (clj->js {:access_token auth-token}))
+                                                            :getBasicProfile
+                                                            (fn [] (clj->js
+                                                                    {:getEmail (fn [] email)
+                                                                     :getId (fn [] sub)}))}))
+                                                :listen (fn [])}
+                                               :signOut
+                                               (fn [] (swap! state update :user-status disj :signed-in))})]
+                                (utils/set-google-auth2-instance! instance)
+                                (swap! state update :user-status conj :signed-in))
+                              (utils/log "Failed to validate token: " status-code)))}))))
```

Comment:
Do you check for log messages like this? I'm worried this will never be seen.

Reply:
I was going to say that I don't really care much about raising visibility because I'd expect a developer to be in the console while debugging a problem like this anyway, but that's not the case if this happens sporadically during an automated test run. I could render an error modal that would be captured by a test failure screenshot if that sounds good to you.

Reply:
I use the console extensively for debugging, so keeping noise out is essential. That's my main concern. Use your own judgement.